service/src/main/kotlin/no/nav/su/se/bakover/service/behandling/BehandlingServiceImpl.kt

```diff
@@ -1,361 +0,0 @@
-                        OppgaveConfig.Saksbehandling(
-                            journalpostId = journalpostId,
-                            søknadId = behandling.søknad.id,
-                            aktørId = aktørId,
-                            tilordnetRessurs = behandling.saksbehandler()
-                        )
-                    ).getOrElse {
-                        log.error("Behandling $behandlingId ble ikke underkjent. Klarte ikke opprette behandlingsoppgave")
-                        return@underkjenn KunneIkkeUnderkjenneBehandling.KunneIkkeOppretteOppgave.left()
-                    }.also {
-                        behandlingMetrics.incrementUnderkjentCounter(UnderkjentHandlinger.OPPRETTET_OPPGAVE)
-                    }
-                    behandling.oppdaterOppgaveId(nyOppgaveId)
-                    behandlingRepo.oppdaterAttestering(behandlingId, attestering)
-                    behandlingRepo.oppdaterOppgaveId(behandling.id, nyOppgaveId)
-                    behandlingRepo.oppdaterBehandlingStatus(it.id, it.status())
-                    log.info("Behandling $behandlingId ble underkjent. Opprettet behandlingsoppgave $nyOppgaveId")
-                    hendelsesloggRepo.oppdaterHendelseslogg(it.hendelseslogg)
-                    behandlingMetrics.incrementUnderkjentCounter(UnderkjentHandlinger.PERSISTERT)
-                    oppgaveService.lukkOppgave(eksisterendeOppgaveId)
-                        .mapLeft {
-                            log.error("Kunne ikke lukke attesteringsoppgave $eksisterendeOppgaveId ved underkjenning av behandlingen. Dette må gjøres manuelt.")
-                        }.map {
-                            log.info("Lukket attesteringsoppgave $eksisterendeOppgaveId ved underkjenning av behandlingen")
-                            behandlingMetrics.incrementUnderkjentCounter(UnderkjentHandlinger.LUKKET_OPPGAVE)
-                        }
-                    behandling.also {
-                        observers.forEach { observer -> observer.handle(Event.Statistikk.BehandlingAttesteringUnderkjent(it)) }
-                    }
-                }
-        }
-    }
-
-    // TODO need to define responsibilities for domain and services.
-    override fun oppdaterBehandlingsinformasjon(
-        behandlingId: UUID,
-        saksbehandler: NavIdentBruker.Saksbehandler,
-        behandlingsinformasjon: Behandlingsinformasjon
-    ): Either<KunneIkkeOppdatereBehandlingsinformasjon, Behandling> {
-        val behandling = behandlingRepo.hentBehandling(behandlingId)
-            ?: return KunneIkkeOppdatereBehandlingsinformasjon.FantIkkeBehandling.left()
-
-        return behandling.oppdaterBehandlingsinformasjon(
-            saksbehandler,
-            behandlingsinformasjon
-        ) // invoke first to perform state-check
-            .mapLeft {
-                KunneIkkeOppdatereBehandlingsinformasjon.AttestantOgSaksbehandlerKanIkkeVæreSammePerson
```

Comment:
Fair å at vi har kvittet oss med denne sjekken for hvert steg. Bør holde at denne sjekken utføres når behandlingen evt sendes til attestering for andre gang. Bør dobbeltsjekke at vi har dekning for det caset.

Reply:
Den eneste plassen vi nå sammenligner saksbehandler og attestant er i selve attestering/iverksettings-steget. Vi legger inn et akseptansetestpunkt på dette så fag får tatt en ekstra vurdering.